pdbtools/pdb_delinsertion.py

```diff
@@ -132,7 +132,7 @@ def delete_insertions(fhandle, option_list):
     for line in fhandle:
 
         if line.startswith(records):
-            res_uid = line[17:26]  # resname, chain, resid
+            res_uid = line[17:27]  # resname, chain, resid, icode
```

Comment:
The proposed change `26` to `27` solves the reported issue but creates another bug that the tests to do currently cover. The proposed implementation works if the `TER` line is also an `insertion`, if not, the current code will remove the `\n` and output the following:

```plain
ATOM      3  HB3 ALA B  10      35.847  44.180   3.181  1.00  0.00           H  
ATOM      3  C   ALA B  10      36.213  46.067   1.258  1.00  0.00           C  
ATOM      3  O   ALA B  10      36.287  47.046   2.028  1.00  0.00           O  
TER       3      ALA B  11 ATOM      3  N   ASN A   1      22.066  40.557   0.420  1.00  0.00           N  
ATOM      3  H   ASN A   1      21.629  41.305  -0.098  1.00  0.00           H  
ATOM      3  H2  ASN A   1      23.236  40.798   0.369  1.00  0.00           H  
```

this can be fixed easily with an additional if-statement, maybe can be fixed hardly in another way that avoids the if

```python
if line.startswith(records):                                            
 10             if line.startswith('TER'):                                          
 11                 res_uid = f'{line[17:26]} '  # resname, chain, resid, icode     
 12             else:                                                               
 13                 res_uid = line[17:27]  # resname, chain, resid, icode           
```

Reply:
The `res_uid` is a variable used to check we are looking at a new residue or not, it does not influence how the lines are written. To avoid this kind of problems, we don't actually touch the lines at all (e.g. `.strip`) and simply cut/paste the bits we need to change, see below.

```
            if clean_icode:  # remove icode
                line = line[:26] + ' ' + line[27:]

            # Modify resid if necessary
            resid = int(line[22:26]) + offset
            line = line[:22] + str(resid).rjust(4) + line[26:]
            seen_ids.add(id_res)
```